scripts/griddle.jsx

```diff
@@ -649,8 +650,13 @@ var Griddle = React.createClass({
         }
     },
     _toggleSelectAll: function () {
-        var visibleRows = this.getDataForRender(this.getCurrentResults(), this.columnSettings.getColumns(), true),
-            newIsSelectAllChecked = !this.state.isSelectAllChecked,
+        var visibleRows = [];
+        if (this.props.enableInfiniteScroll) {
+            visibleRows = this.getCurrentResults();
```

Comment:
It depends on whether we need custom sort for some columns and if all cells are text or can be other components. If so then we need to change code here and add two more features to table.